bitswap/client/internal/sessionmanager/sessionmanager.go

```diff
@@ -153,6 +153,11 @@ func (sm *SessionManager) GetNextSessionID() uint64 {
 
 // ReceiveFrom is called when a new message is received
 func (sm *SessionManager) ReceiveFrom(ctx context.Context, p peer.ID, blks []cid.Cid, haves []cid.Cid, dontHaves []cid.Cid) {
+	// Keep only the keys that at least one session wants
+	keys := sm.sessionInterestManager.FilterInterests(blks, haves, dontHaves)
+	blks = keys[0]
+	haves = keys[1]
+	dontHaves = keys[2]
```

Comment:
Is this already being done redundantly in the call to `sess.ReceiveFrom` below (line 175)?

See: https://github.com/ipfs/boxo/blob/main/bitswap/client/internal/session/session.go#L189